### ISSUE TITLE
Installer fails when upgrading to EE 6.4.2

### ISSUE THREAD

**Issue by ignetic** [ExpressionEngine/ExpressionEngine#2625]
After clicking on the the upgrade from EE 6.3.4 to 6.4.2, I've been presented with the following error during updating files (PHP 7.4):

We stopped on Preflight check.

Fatal error: Uncaught Error: Call to a member function isWritable() on null in /home/path_above_root/system/ee/ExpressionEngine/Boot/boot.common.php:57
Stack trace:
#0 /home/path_above_root/system/ee/legacy/libraries/Log.php(31): is_really_writable()
#1 /home/path_above_root/system/ee/ExpressionEngine/Boot/boot.common.php(118): EE_Log->__construct()
#2 /home/path_above_root/system/ee/ExpressionEngine/Boot/boot.common.php(401): load_class()
#3 /home/path_above_root/system/ee/legacy/core/Utf8.php(26): log_message()
#4 /home/path_above_root/system/ee/ExpressionEngine/Boot/boot.common.php(118): EE_Utf8->__construct()
#5 /home/path_above_root/system/ee/ExpressionEngine/Legacy/App.php(216): load_class()
#6 /home/path_above_root/system/ee/ExpressionEngine/Legacy/App.php(27): ExpressionEngine\Legacy\App->exposeGlobals()
#7 /home/path_above_root/system/ee/Express in /home/path_above_root/system/ee/ExpressionEngine/Boot/boot.common.php on line 57

Simon

**Comment by ignetic** [comment]
This issue also happens with a manual install that I tried.

I managed to get by this by changing the line to:

`if (ee('Filesystem')) {
    return ee('Filesystem')->isWritable($file);
}
return false;`

**Comment by ignetic** [comment]
I've also tested this on another server with the following:

- Upgrading from EE 6.3.5 to 6.4.2
- PHP 7.4.33
- FPM/FastCGI
- Apache 2
- MariaDB 10.3
- Installed addons: CartThrob Pro, CartThrob Order Manager, Rich Text Editor, Statistics

**The error is:**

Oops, looks like the updater couldn't complete.

We stopped on Preflight check.

Fatal error: Uncaught Error: Call to a member function isWritable() on null in /home/ee6inetwebhostco/system/ee/ExpressionEngine/Boot/boot.common.php:57
Stack trace:
#0 /home/ee6inetwebhostco/system/ee/legacy/libraries/Log.php(31): is_really_writable('/home/ee6inetwe...')
#1 /home/ee6inetwebhostco/system/ee/ExpressionEngine/Boot/boot.common.php(118): EE_Log->__construct()
#2 /home/ee6inetwebhostco/system/ee/ExpressionEngine/Boot/boot.common.php(401): load_class('Log')
#3 /home/ee6inetwebhostco/system/ee/legacy/core/Utf8.php(26): log_message('debug', 'Utf8 Class Init...')
#4 /home/ee6inetwebhostco/system/ee/ExpressionEngine/Boot/boot.common.php(118): EE_Utf8->__construct()
#5 /home/ee6inetwebhostco/system/ee/ExpressionEngine/Legacy/App.php(216): load_class('Utf8', 'core')
#6 /home/ee6inetwebhostco/system/ee/ExpressionEngine/Legacy/App.php(27): ExpressionEngine\Legacy\App->exposeGlobals()
#7 /home/ee6inetwebhostco/system/ee/ExpressionEngine/Core/Core.php(366): ExpressionEngine\Legacy\App->boot()
#8 /home/ee6inetwebhostco/sy in /home/ee6inetwebhostco/system/ee/ExpressionEngine/Boot/boot.common.php on line 57

**Comment by ignetic** [comment]
EE7 has this in boot.common.php which may resolve the updating issue in version 6.4.2:
```

        // this may be called before the application core is booted so ee('Filesystem') may not always resolve
        return (new Filesystem)->isWritable($file);

```
EE 6.4.2 currently has:
`
        return ee('Filesystem')->isWritable($file);
`

**Comment by intoeetive** [comment]
Closing as it has been resolved in 6.4.3
